core/src/main/java/org/mifos/mobilewallet/core/domain/usecase/standinginstruction/CreateStandingTransaction.kt

```diff
@@ -0,0 +1,191 @@
+
+                    }
+
+                    override fun onError(e: Throwable) {
+                        useCaseCallback.onError(Constants.ERROR_FETCHING_FROM_ACCOUNT)
+                    }
+
+                    override fun onNext(clientAccounts: ClientAccounts) {
+                        val accounts = clientAccounts.savingsAccounts
+                        if (accounts != null && accounts.size != 0) {
+                            var walletAccount: SavingAccount? = null
+                            for (account in accounts) {
+                                if (account.productId ==
+                                        Constants.WALLET_ACCOUNT_SAVINGS_PRODUCT_ID) {
+                                    walletAccount = account
+                                    break
+                                }
+                            }
+                            walletAccount?.let{
+                                fromAccount = walletAccount
+                                fetchToAccountDetails()
+                            }?: useCaseCallback.onError(Constants.NO_WALLET_FOUND)
+
+                        } else {
+                            useCaseCallback.onError(Constants.ERROR_FETCHING_FROM_ACCOUNT)
+                        }
+                    }
+                })
+    }
+
+    private fun fetchToAccountDetails() {
+        apiRepository.getAccounts(requestValues.toClientId)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(object : Subscriber<ClientAccounts>() {
+                    override fun onCompleted() {}
+                    override fun onError(e: Throwable) {
+                        useCaseCallback.onError(Constants.ERROR_FETCHING_TO_ACCOUNT)
+                    }
+
+                    override fun onNext(clientAccounts: ClientAccounts) {
+                        val accounts = clientAccounts.savingsAccounts
+                        if (accounts != null && accounts.size != 0) {
+                            var walletAccount: SavingAccount? = null
+                            for (account in accounts) {
+                                if (account.productId ==
+                                        Constants.WALLET_ACCOUNT_SAVINGS_PRODUCT_ID) {
+                                    walletAccount = account
+                                    break
+                                }
+                            }
+                            walletAccount?.let{
+                                toAccount = walletAccount
+                                createNewStandingInstruction()
+                            }?: useCaseCallback.onError(Constants.NO_WALLET_FOUND)
+
+                        } else {
+                            useCaseCallback.onError(Constants.ERROR_FETCHING_TO_ACCOUNT)
+                        }
+                    }
+                })
+    }
+
+    private fun createNewStandingInstruction() {
```

Comment:
@devansh-299 It would be much better if we use factory for initializing/creating new standingInstructions throughout the codebase.
Why?
* As it would reduce the following kind/type of repititive code in the concerned file(s) where it is needed to create a new standingInstruct'n

Reply:
@s-ayush2903 I get what that factory pattern will reduce the boilerplate code but we just need this code at two places, hence creating a factory for this in my opinion will be an overkill

Reply:
I'm astonished at consideration of factory as an _overkill_.Let me provide a few more instances which clicked right now(while authoring this thread)where this can be helpful
* Consider the scenario where one is trying to test this feature via api calls first;
* Then as for instruction creation takes 3 mandatory parameters, out of which vpa needs to be validated, so for end-to-end testing(covering all edge cases) one'll have to create two `SIs`-
a valid one and an invalid one,
* If we consider the unit test it solely(not networking calls, just its functionality), that'll also require same two more instances of the same
* So considering this minimum applicability/usages of SIs, the total locs would be=
  `41x(2+2+2)=246` {41=179-136} 
If we had factory then, it would be:- `41x1 + 2+2+2 = 47`

But still, i was just putting forward the standard software design pattern and my invaluable view regarding the same, everything is upto you :-)

Reply:
@s-ayush2903 thank you providing such an in depth analysis and for staging your views. I will consult my mentors over this but anyways I appreciate your thoughtfulness